src/Query/Group.ts

```diff
@@ -166,12 +166,18 @@ export class Group {
             return ['Unknown Location'];
         }
 
-        // Markdown characters in the file name must be escaped.
-        // Markdown characters in the heading must NOT be escaped.
-        const filenameComponent = Group.groupByFileName(task)[0];
+        let filenameComponent = 'Unknown Location';
+
+        if (task.filename !== null) {
+            // Markdown characters in the file name must be escaped.
+            filenameComponent = Group.escapeMarkdownCharacters(task.filename);
+        }
```

Comment:
I see that this was trickier than I expected, in that it required copying-and-pasting some code from `groupByFileName()`.

Ordinarily I would suggest refactoring the code to avoid copy-and-paste, but there is another pull request in progress that this might possibly create conflicts for.

So I will happily accept this as a definite improvement for users!